patches/chrome-browser-ui-omnibox-chrome_omnibox_client.h.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/chrome/browser/ui/omnibox/chrome_omnibox_client.h b/chrome/browser/ui/omnibox/chrome_omnibox_client.h
+index db1e94ddecdefe05ef45d090722eeee3e5eb0105..d79e108c47a13155f0f0e558564d1baf29b7606c 100644
+--- a/chrome/browser/ui/omnibox/chrome_omnibox_client.h
++++ b/chrome/browser/ui/omnibox/chrome_omnibox_client.h
+@@ -17,13 +17,14 @@
+ #include "components/omnibox/browser/favicon_cache.h"
+ #include "components/omnibox/browser/omnibox.mojom-shared.h"
+ #include "components/omnibox/browser/omnibox_client.h"
++#include "brave/components/omnibox/browser/brave_omnibox_client.h"
+ 
+ class Browser;
+ class GURL;
+ class LocationBar;
+ class Profile;
+ 
+-class ChromeOmniboxClient final : public OmniboxClient {
```

Comment:
I'm a bit confused on this, why does the original override break now? Do you possibly just need to add the include for omnibox_client.h in the override before the define?

Reply:
This change is still marked as WIP because I wanted to discuss it during review.

We have to issues here:

 * `ChromeOmniboxClient` is now tagged as `final`, and it would be UB to try to redefine `final` to nothing. We already ha
 * `virtual base::WeakPtr<OmniboxClient> AsWeakPtr() = 0;` has been introduced in the offending CL [1]. The current code on master relies on a substitution of `#define OmniboxClient BraveOmniboxClient` to correct the inheritance for  `ChromeOmniboxClient`, but in the process introducing a substitution in `base::WeakPtr<OmniboxClient> AsWeakPtr() override;`, which breaks the virtual inheritance.

One way to avoid the patching altogether might be doing away entirely with `BraveOmniboxClient`, which only seems to exist as a way to introduce `virtual bool IsAutocompleteEnabled() const;`

Reply:
Simplified in 0e85a99ca65cd5a8a338aa5ba1c2107e48386564.

I have been unable to remove the patch entirely because of the final keyword. But I've done away with `BraveOmniboxClient`, which seemed to exist simply to insert `IsAutocompleteEnabled` in the inheritance chain. It seems to me now that we have a much simpler substitution scheme, by injecting `IsAutocompleteEnabled` directly into the base class `OmniboxClient`, and with that we now can avoid a dangerous downcast operation that was being done from `OmniboxClient` to `BraveOmniboxClient`.

Reply:
I had to revert this simplification @cdesouza-chromium, it was causing all of our browser tests to fail. Let's investigate simplifying this in the scope of cr123. I'll enter a tracking issue for us.

Reply:
https://github.com/brave/brave-browser/issues/35904

Reply:
as mentioned in DM I'm not understanding while `final` is an issue here since we are not subclassing `ChromeOmniboxClient`, we're subclassing `OmniboxClient`

Reply:
I'm not going to block on this, but lets discuss a follow-up